assets/rpip-52/rewards-file-spec.md

```diff
@@ -0,0 +1,102 @@
+# Specification for Rewards Tree File
+
+This documents serves as a formal specification for the format and naming of Rewards Tree Files. 
+These files are generated by the oDAO in an implicit format, prior to this specification. 
+This specification serves to define their expected format, naming convention, and CID calcuation methodology.
+
+# Contents
+
+The oDAO shall save Rewards Trees in [ssz](../assets/rpip-52/simple-serialize.md) format respecting the following specification.
+
+## Containers
+
+### File
+
+```python
+class File(Container):
+    magic: Bytes4 # Magic Header identifying this file as a Rewards Tree
+                  # Expected value: 0x52 0x50 0x52 0x54
+    rewards_file_version: uint64 # Expected value: 3
+    ruleset_version: uint64 # Expected value: 9 or higher
+    network: uint64 # Chain ID for the network, e.g., 1 for Mainnet or 17000 for Holešky
+                    # devnet deploys should use a unique id
+    index: uint64 # Rewards interval index
+    start_time: uint64 # Unix time of the first slot of the interval
+    end_time: uint64 # Unix time of the last slot of the interval
+    consensus_start_block: uint64 # Index of the first non-empty slot of the interval
+    consensus_end_block: uint64 # Index of the last non-empty slot of the interval
+    execution_start_block: uint64 # Index of the execution block in ConsensusStartBlock
+    execution_end_block: uint64 # Index of the execution block in ConsensusEndBlock
+    intervals_passed: uint64 # Number of rewards intervals contained in the file
+    merkle_root: Bytes32 # Merkle Tree root of the Rewards Tree
+    total_rewards: TotalRewards # Aggregate data on values of contained rewards
+    network_rewards: List[NetworkReward, 128] # L1 and L2 rewards destinations and aggregate amounts
+                                              # Sorted ascending by Network
```

Comment:
I guess Network here and Address below shouldn't be capitalised since the field names inside the containers aren't.